epictrack-api/src/api/reports/anticipated_schedule_report.py

```diff
@@ -228,3 +164,95 @@ def generate_report(self, report_date, return_type):
             self.report_title, jsonify(api_payload).json, template
         )
         return report, f"{self.report_title}_{report_date:%Y_%m_%d}.pdf"
+
+    def _get_next_pcp_query(self, start_date):
+        """Create and return the subquery for next PCP event based on start date"""
+        pecp_configuration_ids = (
+            db.session.execute(
+                select(EventConfiguration.id).where(
+                    EventConfiguration.event_category_id == EventCategoryEnum.PCP.value,
+                )
+            )
+            .scalars()
+            .all()
+        )
+        next_pcp_min_date_query = (
+            db.session.query(
+                Event.work_id,
+                func.min(
+                    func.coalesce(Event.actual_date, Event.anticipated_date)
+                ).label("min_pcp_date"),
+            )
+            .filter(
+                func.coalesce(Event.actual_date, Event.anticipated_date) >= start_date,
+                Event.event_configuration_id.in_(pecp_configuration_ids),
+            )
+            .group_by(Event.work_id)
+            .subquery()
+        )
+        next_pecp_query = (
+            db.session.query(
+                Event,
+            )
+            .join(
+                next_pcp_min_date_query,
+                and_(
+                    next_pcp_min_date_query.c.work_id == Event.work_id,
+                    func.coalesce(Event.actual_date, Event.anticipated_date) == next_pcp_min_date_query.c.min_pcp_date,
+                ),
+            )
+            .filter(
+                Event.event_configuration_id.in_(pecp_configuration_ids),
+            )
+            .subquery()
+        )
+        return next_pecp_query
+
+    def _get_referral_event_query(self, start_date):
+        """Create and return the subquery to find next referral event based on start date"""
+        return (
+            db.session.query(
+                Event.work_id,
+                func.min(Event.anticipated_date).label("min_anticipated_date"),
+            )
+            .join(
+                EventConfiguration,
+                and_(
+                    Event.event_configuration_id == EventConfiguration.id,
+                    EventConfiguration.event_type_id == 5,
```

Comment:
Instead of hard coding this number 5, I think we have an enum in the model folder